zerver/views/streams.py

```diff
@@ -631,10 +631,19 @@ def send_messages_for_new_subscribers(
                     content = _("{user_name} created a new stream {stream_str}.")
                 topic = _("new streams")
 
-            content = content.format(
-                user_name=f"@_**{user_profile.full_name}|{user_profile.id}**",
-                stream_str=", ".join(f"#**{s.name}**" for s in created_streams),
-            )
+                stream_strs = map(
+                    lambda s: _("#**{stream_name}** (*{stream_description}*)").format(
+                        stream_name=s.name, stream_description=s.description
+                    )
+                    if s.description != ""
+                    else f"#**{s.name}**",
+                    created_streams,
+                )
+
+                content = content.format(
+                    user_name=f"@_**{user_profile.full_name}|{user_profile.id}**",
+                    stream_str=_(", ").join(stream_strs),
+                )
```

Comment:
I think given the descriptions, the `len(created_streams) > `) case might be better handled as this template:
```
{user_name} created the following streams:
* {stream_str1}
* {stream_str2}
...
```

Since I think with descriptions of moderate length, it'll end up being pretty ugly with just comma separation.  And this also lets us avoid i18n, since we're using a markdown feature for the list instead.

Reply:
Also, there's sorta a messy issue in the more common single-stream case, because if the stream description has a `.` at the end, we're going to have a double-`.` that will look ugly.  Not sure what the best fix is, but I suppose one option is to just use the bulleted list template always?  Might be a little more scannable that way.

Or we could doing a style where every message is:
```
* {stream_name} ({stream_description}) was created by {user_name}.
```

which would both be easier on translators and potentially scan more nicely when there's a lot of them in a row in a topic?

Reply:
Hmmm, yeah. I'm inclined to optimize for the single-stream case, since AFAICT that will be the situation the vast majority of the time new streams are created. I think it's also worth thinking about old messages when considering scanability — keeping the format similar to what it currently is seems valuable for that. Given that, I'll propose:

```
{user_name} created a new stream {stream_name} ({stream description})
```

for the case with the description, and:

```
{user_name} created a new stream {stream_name}.
```

for the case without a description, and multiple lines being handled with linebreaks (since IMO bulletpoints are a little ugly in the single-stream case).

This will keep the format similar to what it currently is, work reasonably well with internationalization, and eliminates the punctuation-on-both-sides-of-parenthesis thing. It might be slightly strange in the case where a stream description doesn't end with punctuation, but I think overall that will not be very noticeable (given that erroneous punctuation is more noticeable than lack of punctuation)

Reply:
Went ahead and changed this PR to the format I suggested above, although I don't really have strong feelings on which one to use, if you're prefer something different

Reply:
>  (since IMO bulletpoints are a little ugly in the single-stream case).

I'm not sure it actually will be ugly, in that it's likely that the topic in question will have sequential messages with the bullets that look like they're all sorta one sequence.  But certainly I'd want to see it (perhaps with some human messages mixed in) before we decided to go with bullets.